library/system/src/lib/yast2/execute.rb

```diff
@@ -87,11 +104,119 @@ def self.locally(*args)
     # {http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast%2FLogger Y2Logger}.
     # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
     # @raise Cheetah::ExecutionFailed if the command fails
-    def self.locally!(*args)
-      Cheetah.run(*args)
+    def locally!(*args)
+      run_or_chain(args)
+    end
+
+    # Runs a command described by *args* and returns its output
+    #
+    # It also globally switches the default Cheetah logger to
+    # {http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast%2FLogger Y2Logger}.
+    #
+    # @param args [Array<Object>] see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
+    # @return [String] command output or an empty string if the command fails.
+    def stdout(*args)
+      chaining_object(yast_stdout: true, stdout: :capture).run_or_chain(args)
+    end
+
+  protected
+
+    # Decides either to run the command or to chain the call in case that no argmuments
+    # are given.
+    #
+    # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
+    # @return [Object, ExecuteClass] result of running the command or a chaining object.
+    def run_or_chain(args)
+      args.none? ? self : run(*args)
+    end
+
+  private
+
+    # Options to add when running a command
+    #
+    # Some options are intended to control the behavior and they are not passed to
+    # Cheetah.run. For example:
```

Comment:
I think this statement is not true as it is actually passed to cheetah, which ignore it.

Reply:
ah, now I see it. OK.

Reply:
"yast_*" options are filtered out, see `#command_options` method.